src/Teams/beta/custom/RscConfigurationConverter.cs

```diff
@@ -0,0 +1,165 @@
+                this.LogVerbose("Chat RSC is enabled in Teams App Settings.", eventListener);
+                microsoftGraphRscConfiguration.State = MicrosoftGraphRscConfigurationState.EnabledForAllApps;
+            }
+            else if (authorizationPolicy
+                ?.DefaultUserRolePermissions
+                ?.PermissionGrantPoliciesAssigned
+                ?.Contains(RscConfigurationConverter.MicrosoftCreatedPermissionGrantPolicyForChatRscPreApproval, StringComparer.OrdinalIgnoreCase) == true)
+            {
+                this.LogVerbose("Authorization policy contains permission grant policy for chat RSC preapprovals.", eventListener);
+                microsoftGraphRscConfiguration.State = MicrosoftGraphRscConfigurationState.EnabledForPreApprovedAppsOnly;
+            }
+            else
+            {
+                this.LogVerbose("Chat RSC is disabled.", eventListener);
+                microsoftGraphRscConfiguration.State = MicrosoftGraphRscConfigurationState.Disabled;
+            }
+
+            return microsoftGraphRscConfiguration;
+        }
+
+        /// <summary>
+        /// Convert the given tenant settings to Team RSC configuration.
+        /// </summary>
+        /// <param name="tenantConsentSettingCollection">Tenant consent setting collection.</param>
+        /// <param name="authorizationPolicy">Authorization policy.</param>
+        /// <returns>Rsc configuration.</returns>
+        internal IMicrosoftGraphRscConfiguration ConvertToTeamRscConfiguration(
+            MGTeamsInternalTenantConsentSettingsCollection tenantConsentSettingCollection,
+            MGTeamsInternalAuthorizationPolicy authorizationPolicy,
+            Runtime.IEventListener eventListener)
+        {
+            if (tenantConsentSettingCollection == null)
+            {
+                throw new MGTeamsInternalException(
+                    MGTeamsInternalErrorType.ResourceNotFound,
+                    "Tenant consent settings were not found.");
+            }
+
+            if (authorizationPolicy == null)
+            {
+                throw new MGTeamsInternalException(
+                    MGTeamsInternalErrorType.ResourceNotFound,
+                    "Authorization policy was not found.");
+            }
+
+            MicrosoftGraphRscConfiguration microsoftGraphRscConfiguration = new MicrosoftGraphRscConfiguration
+            {
+                Id = "TeamResourceSpecificPermissionConfiguration",
+                ScopeType = MicrosoftGraphRscConfigurationScopeType.Team
+            };
+
+            MGTeamsInternalTenantConsentSettings groupConsentSettings = tenantConsentSettingCollection.Value?.FirstOrDefault(
+                v => string.Equals(v.TemplateId, RscConfigurationConverter.GroupConsentSettingsTemplateId, StringComparison.OrdinalIgnoreCase));
+
+            if (groupConsentSettings == null)
+            {
+                this.LogVerbose("Group Consent settings were not found.", eventListener);
```

Comment:
If null, then check the authorization policy to compute the semantics of the null by looking for user consent settings Legacy Default ON.
![image](https://user-images.githubusercontent.com/25969716/234367021-d8196196-fe4f-4e19-a72e-f89056c2dfef.png)

![image](https://user-images.githubusercontent.com/25969716/234367258-f13126f9-4d7e-4e0b-bdc1-2261669e2abc.png)